### ISSUE TITLE
[MaterializedView]Wrong result return when mv was matched

### ISSUE THREAD

**Issue by caneGuy** [StarRocks/starrocks#7274]
### Steps to reproduce the behavior (Required)
We have a table and create mv for it.
```
CREATE TABLE test_bl(
timed DATE, event varchar(30), pid varchar(128), brand varchar(64), userId varchar(256), __time__ BIGINT, timedt DATETIME,
 qSlotId varchar(128), clientIP varchar(256), city_name varchar(64), genUniqId varchar(256), adSlotName varchar(128), adId varchar(256),
oaid varchar(512), fileName STRING, __tag__hostname__ varchar(256), __tag__path__ varchar(256), __tag__receive_time__ varchar(256), __tag__user_defined_id__ varchar(256),
product varchar(128), platformVersionName varchar(64), osVersionCode int, uniqId varchar(256), message STRING, manufacturer varchar(64),
osVersionName varchar(128), hostPackageName varchar(512), placeholder STRING, __source__  varchar(256), source STRING, clientTime varchar(128),
country_name varchar(128), goldVersion varchar(64), model varchar(128), adType varchar(64), imeimd5 varchar(256), userAgent STRING,
redirect_err varchar(512), showDuration varchar(512), qSetId varchar(256), landingURL STRING, fromd varchar(128), __topic__ STRING,
redirect_url STRING, region_name varchar(256), timeHide BIGINT, adData STRING, androididmd5 varchar(256), refer STRING, isProxy varchar(32),
platformVersionCode BIGINT, osType varchar(64), page varchar(256), adConfResult varchar(64), reqPackageName varchar(512), userMode varchar(32),
timeShow BIGINT, oaidmd5 varchar(256), host varchar(300), packageName varchar(128), timeId BIGINT, q STRING, qClickId varchar(800),
addata__adid BIGINT, addata__clickbtntxt varchar(256), addata__creativetype BIGINT, addata__desc STRING, addata__error varchar(512),
addata__error__errcode varchar(64), addata__error__errmsg varchar(1280), addata__icon varchar(1280), addata__imgurllist varchar(512),
addata__interactiontype varchar(32), addata__logourl varchar(512), addata__source varchar(128), addata__title varchar(512),
callback_url varchar(1280), click_device_id varchar(256), closetype varchar(64), complaindetail STRING, complainreason varchar(128),
cost varchar(64), deviceid varchar(64), dpproductname varchar(256), errors varchar(1280), ip varchar(256),level varchar(64), method varchar(64),
path STRING, query STRING, reqip varchar(128), requestid varchar(256), requseragent STRING, reviewertype varchar(1280), status varchar(64),
source__extra varchar(64), source__extra__third_st_param varchar(1280), source__extra__url_h5 varchar(1280), source__packagename varchar(512),
source__type varchar(128), task varchar(128), scene varchar(128), title varchar(512), cur_rate varchar(64), tab STRING, confAdId varchar(256), confAdIds ARRAY<VARCHAR(256)>, confTag VARCHAR(128), confName VARCHAR(128), timeh INT
)
DUPLICATE KEY(timed, event) 
PARTITION BY RANGE(timed)(
START ("2022-06-10") END ("2022-06-31") EVERY (INTERVAL 1 day)
)
DISTRIBUTED BY HASH(userId) BUCKETS 64
PROPERTIES(

"dynamic_partition.enable" = "true",
"dynamic_partition.time_unit" = "DAY",
"dynamic_partition.start" = "-90",
"dynamic_partition.end" = "45",
"dynamic_partition.prefix" = "p",
"dynamic_partition.buckets" = "64"

);
```
```
CREATE MATERIALIZED VIEW test_bl_mvttepbr AS
select timed, timeh, event, pid, brand, redirect_err, count(timedt) as PV
from test_bl
group by timed, timeh, event, pid, brand, redirect_err

CREATE MATERIALIZED VIEW test_bl_mvttpc AS
select timed, timeh, event, pid, city_name, count(timedt) as PV
from test_bl
group by timed, timeh, event, pid, city_name
```

### Expected behavior (Required)
get the same result for count(xx) and count(1)

### Real behavior (Required)
For count(1):
```
select
    timed,
    timeh,
    sum(epcount)
from 
(
    select 
        timed, timeh, pid, brand, count(1) as epcount
    from 
        test_bl
    where 
        timed = current_date()
    group by 
        timed, timeh, pid, brand    
) tp
where 
    pid not like 'API%' 
    and pid not like 'tt-%' and pid not like 'szxd%' 
    and pid not like 'dswl%' and pid not like 'kksp%'
    and pid not like 'rta-%'
group by timed, timeh
order by timed, timeh
```
![image](https://user-images.githubusercontent.com/26762018/173765076-8545f30c-fa44-4acc-b489-e2f24b3ce841.png)

For count(timedt)
```
select
    timed,
    timeh,
    sum(epcount)
from 
(
    select 
        timed, timeh, pid, brand, count(timedt) as epcount
    from 
        test_bl
    where 
        timed = current_date()
    group by 
        timed, timeh, pid, brand    
) tp
where 
    pid not like 'API%' 
    and pid not like 'tt-%' and pid not like 'szxd%' 
    and pid not like 'dswl%' and pid not like 'kksp%'
    and pid not like 'rta-%'
group by timed, timeh
order by timed, timeh
```
![image](https://user-images.githubusercontent.com/26762018/173765186-18d4740e-300a-4787-b432-acd5d4b5a04b.png)

Plan:
for count(1)

[5E4415A4-F1FF-43ef-AC18-952DC97AD1F4.txt](https://github.com/StarRocks/starrocks/files/8906733/5E4415A4-F1FF-43ef-AC18-952DC97AD1F4.txt)

for count(timedt) we hit the mv rollup
[39103E21-5B11-4b1d-9B4A-F08E1B1914A7.txt](https://github.com/StarRocks/starrocks/files/8906734/39103E21-5B11-4b1d-9B4A-F08E1B1914A7.txt)

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.1.8

**Comment by caneGuy** [comment]
![image](https://user-images.githubusercontent.com/26762018/173766884-3f3ae7ab-b224-47b9-8292-3d6f360909f6.png)

**Comment by caneGuy** [comment]
![截屏2022-06-16 上午10 01 47](https://user-images.githubusercontent.com/26762018/173975196-9f8f16b8-5af2-4490-9af0-5a4fb9a1b936.png)

count(timedt)
[F3E63AB5-6CCD-48e3-A18F-66001889406D.txt](https://github.com/StarRocks/starrocks/files/8915053/F3E63AB5-6CCD-48e3-A18F-66001889406D.txt)

count(1)
[182DEC15-0DBC-435b-BE29-9169B2F3EFD9.txt](https://github.com/StarRocks/starrocks/files/8915052/182DEC15-0DBC-435b-BE29-9169B2F3EFD9.txt)

**Comment by caneGuy** [comment]
I reproduced this issue with 2.1.8
- create table
```
CREATE TABLE test_bl_1(
timed DATE, event varchar(30), pid varchar(128), brand varchar(64), userId varchar(256), timedt DATETIME, timeh INT
)
DUPLICATE KEY(timed, event) 
PARTITION BY RANGE(timed)(
START ("2022-06-10") END ("2022-06-30") EVERY (INTERVAL 1 day)
)
DISTRIBUTED BY HASH(userId) BUCKETS 64
PROPERTIES(

"dynamic_partition.enable" = "true",
"dynamic_partition.time_unit" = "DAY",
"dynamic_partition.start" = "-90",
"dynamic_partition.end" = "45",
"dynamic_partition.prefix" = "p",
"dynamic_partition.buckets" = "64"

);
```
- create mv
```
CREATE MATERIALIZED VIEW test_bl_mvttepbr AS
select timed, timeh, event, pid, brand, count(timedt) as PV
from test_bl
group by timed, timeh, event, pid, brand
```
- insert one record
```
insert into test_bl_1 values("2022-6-16", "test1", "yz", "testb", "zk", "2022-6-06 06:00:00", 6);
```
- test with

![image](https://user-images.githubusercontent.com/26762018/174736679-daf54a54-31a0-42db-b1ba-696eb8c2ab06.png)

```
select     timed,     timeh,     sum(epcount) from  (     select          timed, timeh, pid, brand, count(timedt) as epcount     from          test_bl     where          timed = "2022-06-16"     group by          timed, timeh, pid, brand     ) tp where      pid not like 'API%'      and pid not like 'tt-%' and pid not like 'szxd%'      and pid not like 'dswl%' and pid not like 'kksp%'     and pid not like 'rta-%' group by timed, timeh order by timed, timeh;
```

**Comment by caneGuy** [comment]
query below will return wrong result
```
select   sum(tt) from  (     select          timed, timeh, pid, brand, count(timedt) as tt     from          test_bl    where          timed = "2022-06-16"     group by          timed, timeh, pid, brand     ) tp;
```
query below will return right result
```
select          timed, timeh, pid, brand, count(timedt) as tt     from          test_bl    where          timed = "2022-06-16"     group by          timed, timeh, pid, brand
```

**Comment by kangkaisen** [comment]
fixed in 2.2 version
